src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -75,11 +76,23 @@ public RepoConfiguration(RepoLocation location, String branch, List<Format> form
     }
 
     public static void setDatesToRepoConfigs(
```

Comment:
Under the situation now, do we really need a `sinceDate` and `untilDate` in each `RepoConfiguration`?

Reply:
I believe it's required. When extracting commit information (in line 33): -

https://github.com/reposense/RepoSense/blob/27655c76427fd97a00b3ee2debcd348c8b94b047/src/main/java/reposense/commits/CommitInfoExtractor.java#L22-L33

`git log` will be called, which takes in `since date` and `until date` for each repo configurations: -
https://github.com/reposense/RepoSense/blob/27655c76427fd97a00b3ee2debcd348c8b94b047/src/main/java/reposense/git/GitLog.java#L17-L30

If `since date` or `until date` not provided, the first commit date and last commit date will be set respectively, which is what this PR is trying to address.

Reply:
What I mean previously is to use cliSince and cliUntil directly when analyzing the repo. Because now, all the repoconfig will have same since and same until date. However, as the discussion below, we may still need them to be separated. So I will resolve the conversation.